tests/test_manager.py

```diff
@@ -138,6 +138,57 @@ def test_save_to_folders(tmpdir, manager, result, parameter_combination_range):
                             str(tmpdir.join('folder_export')),
                             2)
 
+def test_run_logging_simulation(ns_3_compiled_debug,config):        #TODO cannot find the scripts in /examples folder(even if --enable-examples is passed in the ns_3_compiled_debug fixture in conftest.py). Why does this happen?
+    params = {
+        'powerLevels': 18.0,
+        'manager': 'ns3::ParfWifiManager',
+        'AP1_y': 0,
+        'STA1_y': 0,
+        'maxPower': 17,
+        'outputFileName': 'parf',
+        'AP1_x': 0,
+        'STA1_x': 5.0,
+        'stepsTime': 1,
+        'minPower': 0,
+        'steps':200,
+        'rtsThreshold': 2346,
+        'stepsSize':1
+    }
+    log_component = {
+        'PowerAdaptationDistance':'debug'
+    }
+    new_campaign = sem.CampaignManager.new(ns_3_compiled_debug,
+                                            'wifi-power-adaptation-distance',
+                                            config['campaign_dir'],
+                                            overwrite=True)
+
+    #Ensure that the simulations run fine                                            
+    log_path = new_campaign.run_missing_simulations(params,log_component=log_component)                                        
+
+    #Assert log_path list contains exctly one entry as only one parameter 
+    # combination is passed
+    assert len(log_path) == 1
+
+    #Assert the log file generated is correct
+
+    sample_logs = ['+0.000000000s -1 PowerAdaptationDistance:SetupPhy(): [DEBUG] OfdmRate6Mbps 0.00192 6000000bps'
+                    '+0.000000000s -1 PowerAdaptationDistance:SetupPhy(): [DEBUG] OfdmRate9Mbps 0.001288 9000000bps',
+                    '+0.000000000s -1 PowerAdaptationDistance:SetupPhy(): [DEBUG] OfdmRate12Mbps 0.000972 12000000bps',
+                    '+0.000000000s -1 PowerAdaptationDistance:SetupPhy(): [DEBUG] OfdmRate18Mbps 0.000656 18000000bps',
+                    '+0.000000000s -1 PowerAdaptationDistance:SetupPhy(): [DEBUG] OfdmRate24Mbps 0.000496 24000000bps',
+                    '+0.000000000s -1 PowerAdaptationDistance:SetupPhy(): [DEBUG] OfdmRate36Mbps 0.00034 36000000bps',
+                    '+0.000000000s -1 PowerAdaptationDistance:SetupPhy(): [DEBUG] OfdmRate48Mbps 0.00026 48000000bps',
+                    '+0.000000000s -1 PowerAdaptationDistance:SetupPhy(): [DEBUG] OfdmRate54Mbps 0.000232 54000000bps'
+                ]
+
+    with open(log_path[0]) as f:
+        logs = f.readlines
+    actual_logs = logs.split('\n')
+
+    assert (sample_logs == actual_logs)
```

Comment:
it's quite unstable. if ns-3 is updated (e.g., ns-3.34 will heavily change the wifi module), this test will fail.
as sem supports a multiple ns-3 versions, this check is bound to fail eventually.

also running an ns-3 example is not a great a idea for the same reason. i'd rather suggest to write a simple ns-3 script outputting some pre-defined logs, and check those.

more tests are needed to cover all the updates brought by the additional code. did you check the test coverage before and after your updates? to improve the sem project, coverage should increase or stay the same

Reply:
I agree that using this as a check is a bit dangerous. It's true we have a fixed version of ns-3 we use to run tests and examples, but it's also true that I would like this to be updated every once in a while, and having a heavy dependence on the behavior of a specific ns-3 example could be a headache to maintain.

Writing a custom ns-3 script to test this sounds like a good idea!

Reply:
that's actually a good point. i don't have a clear answer right now, we should discuss it tomorrow in our call

Reply:
As of now, for both examples and tests we rely on a fork of ns-3-dev hosted on my github (the `sem-examples` and `sem-tests` branches [here](https://github.com/dvdmgr/ns-3-dev)). I can give @akshitpatel01 access to this repository for now, and we can think of a better place to host them later (for instance, on the signetlabdei organization page).